src/parsec.rs

```diff
@@ -296,14 +311,27 @@ impl ParsecMap {
 
 // Generate a DkgResult that can be used for the first node
 pub fn generate_first_dkg_result(rng: &mut MainRng) -> DkgResult {
-    let threshold = 0;
+    let participants = 1;
     let index = 0;
 
-    let secret_key_set = RealBlsSecretKeySet::random(threshold, &mut RngCompat(rng));
+    let secret_key_set = generate_bls_threshold_secret_key(rng, participants);
     let secret_key_share = secret_key_set.secret_key_share(index);
     DkgResult::new(secret_key_set.public_keys(), Some(secret_key_share))
 }
 
+/// Generate a BLS SecretKeySet for the given number of participants.
+/// Used for generating first node, or for test.
+pub fn generate_bls_threshold_secret_key(
+    rng: &mut MainRng,
+    participants: usize,
+) -> RealBlsSecretKeySet {
+    // The BLS scheme will require more than `participants / 3`
+    // shares in order to construct a full key or signature.
+    let threshold = participants.saturating_sub(1) / 3;
```

Comment:
Why do yo subtract 1?

Reply:
It is from Parsec.
```
/// Threshold to use for running DKG
pub fn dkg_threshold(participants_count: usize) -> usize {
    participants_count.saturating_sub(1) / 3
}
```

I tried to think about it, I think the reason we substract 1, is because for the example of 3, we need 2*t + 1 answer from the DKG to complete, that would mean 3 if threshold was 1, but then that would mean all answered and that cannot be the case.

For the example of 4, 2t+1 = 3, t = 1 invalid, which may not have answered.
So I think for number divisible by 3, the would be to big. (Sorry, a bit hand wavy, I think I and @fizyk20 had thought through it a bit more when we did the DKG in parsec, I guess the key_gen doc in parsec is probably the best source of info).

Reply:
OK, makes sense.